core/src/test/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoHashGridParserTests.java

```diff
@@ -46,19 +46,6 @@ public void testParseValidFromStrings() throws Exception {
         assertNotNull(GeoGridAggregationBuilder.parse("geohash_grid", stParser));
     }
 
-    public void testParseErrorOnNonIntPrecision() throws Exception {
```

Comment:
I removed this test since this change breaks it _but_ I don't think that this test makes sense in the current state. For example, before this change that test would also break if you changed it from `"2.0"` to a decimal literal like `2.5`. Now it is consistent (it will accept and truncate decimals, whether you quote them or not).